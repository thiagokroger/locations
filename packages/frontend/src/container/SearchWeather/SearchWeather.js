import React from 'react'
import PropTypes from 'prop-types'

const SearchWeather = ({ zipCode, setZipCode, searchPlace, isLoading }) => {
  return (
    <div className="field has-addons has-addons-centered">
      <div className="control has-icons-left">
        <input
          name="search"
          className="input is-rounded"
          type="text"
          placeholder="Address"
          value={zipCode}
          onChange={(e) => setZipCode(e.target.value)}
          disabled={isLoading}
        />
        <span className="icon is-left">
          <i className="fas fa-globe"></i>
        </span>
      </div>
      <div className="control" >
        <button onClick={searchPlace} className={`button is-info ${isLoading ? 'is-loading' : ''}`}>Search</button>
      </div>
    </div>
  )
}

SearchWeather.propTypes = {
  zipCode: PropTypes.string.isRequired,
  setZipCode: PropTypes.func.isRequired,
  searchPlace: PropTypes.func.isRequired,
  isLoading: PropTypes.bool.isRequired
}

export default SearchWeather
