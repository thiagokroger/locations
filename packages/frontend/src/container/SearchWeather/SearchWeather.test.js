import React from 'react'
import { shallow } from 'enzyme'
import SearchWeather from './SearchWeather'

describe('Containers - Past Weather Column without isLoading', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <SearchWeather
        zipCode=""
        setZipCode={jest.fn()}
        searchPlace={jest.fn()}
        isLoading={false}
      />
    )
    expect(component).toMatchSnapshot()
  })

  it('should trigger input correctly', () => {
    const mockFunc = jest.fn()
    const component = shallow(
      <SearchWeather
        zipCode=""
        setZipCode={mockFunc}
        searchPlace={jest.fn()}
        isLoading={false}
      />
    )

    component.find('input').simulate('change', { target: { value: '' } })

    expect(mockFunc.mock.calls.length).toBe(1)
    expect(component).toMatchSnapshot()
  })

  it('should trigger button correctly', () => {
    const mockFunc = jest.fn()

    const component = shallow(
      <SearchWeather
        zipCode=""
        setZipCode={jest.fn()}
        searchPlace={mockFunc}
        isLoading={false}
      />
    )

    component.find('button').simulate('click')

    expect(mockFunc.mock.calls.length).toBe(1)
    expect(component).toMatchSnapshot()
  })
})

describe('Containers - Past Weather Column with isLoading', () => {
  it('should render without crashing', () => {
    const component = shallow(
      <SearchWeather
        zipCode=""
        setZipCode={jest.fn()}
        searchPlace={jest.fn()}
        isLoading={true}
      />
    )
    expect(component).toMatchSnapshot()
  })
})
