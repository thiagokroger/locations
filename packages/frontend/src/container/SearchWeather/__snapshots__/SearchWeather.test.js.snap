// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Containers - Past Weather Column with isLoading should render without crashing 1`] = `
<div
  className="field has-addons has-addons-centered"
>
  <div
    className="control has-icons-left"
  >
    <input
      className="input is-rounded"
      disabled={true}
      name="search"
      onChange={[Function]}
      placeholder="Zip code"
      type="text"
      value=""
    />
    <span
      className="icon is-left"
    >
      <i
        className="fas fa-globe"
      />
    </span>
  </div>
  <div
    className="control"
  >
    <button
      className="button is-info is-loading"
      onClick={[MockFunction]}
    >
      Search
    </button>
  </div>
</div>
`;

exports[`Containers - Past Weather Column without isLoading should render without crashing 1`] = `
<div
  className="field has-addons has-addons-centered"
>
  <div
    className="control has-icons-left"
  >
    <input
      className="input is-rounded"
      disabled={false}
      name="search"
      onChange={[Function]}
      placeholder="Zip code"
      type="text"
      value=""
    />
    <span
      className="icon is-left"
    >
      <i
        className="fas fa-globe"
      />
    </span>
  </div>
  <div
    className="control"
  >
    <button
      className="button is-info "
      onClick={[MockFunction]}
    >
      Search
    </button>
  </div>
</div>
`;

exports[`Containers - Past Weather Column without isLoading should trigger button correctly 1`] = `
<div
  className="field has-addons has-addons-centered"
>
  <div
    className="control has-icons-left"
  >
    <input
      className="input is-rounded"
      disabled={false}
      name="search"
      onChange={[Function]}
      placeholder="Zip code"
      type="text"
      value=""
    />
    <span
      className="icon is-left"
    >
      <i
        className="fas fa-globe"
      />
    </span>
  </div>
  <div
    className="control"
  >
    <button
      className="button is-info "
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      Search
    </button>
  </div>
</div>
`;

exports[`Containers - Past Weather Column without isLoading should trigger input correctly 1`] = `
<div
  className="field has-addons has-addons-centered"
>
  <div
    className="control has-icons-left"
  >
    <input
      className="input is-rounded"
      disabled={false}
      name="search"
      onChange={[Function]}
      placeholder="Zip code"
      type="text"
      value=""
    />
    <span
      className="icon is-left"
    >
      <i
        className="fas fa-globe"
      />
    </span>
  </div>
  <div
    className="control"
  >
    <button
      className="button is-info "
      onClick={[MockFunction]}
    >
      Search
    </button>
  </div>
</div>
`;
